sled-storage/src/manager.rs

```diff
@@ -1996,6 +2059,111 @@ mod tests {
         logctx.cleanup_successful();
     }
 
+    #[tokio::test]
+    async fn ensure_many_datasets() {
```

Comment:
I've been running this test with the following d script:

```d
#pragma D option quiet

/* BEGIN - Tracking process execution */
proc:::exec-success
{
    /* Record the process start time and arguments */
    self->start_time = timestamp;
    printf("Started[%d] %s\n", pid, curpsinfo->pr_psargs);
/*    system("pargs %d; prun %d", pid, pid); */
}

/* BEGIN - Tracking process exit */
proc:::exit
/self->start_time != 0/
{
    /* Calculate elapsed time in milliseconds */
    this->elapsed_ms = (timestamp - self->start_time) / 1000000;

    /* Print process information */
    printf("Finished[%d] in %d ms\n", pid, this->elapsed_ms);

    /* Clear the recorded data */
    self->start_time = 0;
    self->cmdline = 0;
}
```

To monitor "what processes are launched by this test, and how long do they take".

That's where I noticed unnecessary work on `zfs set`, that I optimized.

Reply:
Nice!